### DISCUSSION TITLE
Subscribing a user to multiple topics

### DISCUSSION THREAD

**Discussion by sujeet-agrahari** [kreait/firebase-php#330] [Ideas]
I searched for subscribing a user to multiple topics, but I couldn't find any reference [here](https://firebase-php.readthedocs.io/en/4.32.0/cloud-messaging.html#subscribe-to-a-topic)

I want to subscribe a user to multiple topics something like below - 
```
$topics = ['topicA', 'topicB', 'topicC' ...];
$token = 'user_registeration_token_here';

$firebase
    ->getMessaging()
    ->subscribeToTopic($topics, $token);
```
I tried, but got below error.
> Message: Argument 1 passed to Kreait\Firebase\Messaging\Topic::fromValue() must be of the type string, array given, called in C:\wamp64\www\uat_api\vendor\kreait\firebase-php\src\Firebase\Messaging.php on line 152

Is it possible to do it Or I have to call the method multiple times to subscribe to  each topic?

Will it affect performance If I go with calling the multiple times using for loop?
```
$token = 'user_registeration_token_here';
$topics = $topics = ['topicA', 'topicB', 'topicC' ...];
foreach($topics as $topic) {
    $messaging->subscribeToTopic($topic, $token);
}
```

**Comment by jeromegamez** [comment]
Looping over the topics as you described is the way to go for now, but I‘ll consider this a feature request for a later release ^^

**Comment by sujeet-agrahari** [comment]
@jeromegamez  One more thing, I know it's off topic. I face an issue for variable not set FIREBASE_CRENDENTIALS.

> Type: Kreait\Firebase\Exception\ServiceAccountDiscoveryFailed

 I  am using this package in codeigniter. I am using below method due to this reason. 
`$firebase = (new Factory) ->withServiceAccount('/path/to/google-service-account.json')`


I tried setting a `$autoload['config'] = array('firebase');` in `autoload.php`.
This what I have in my `config/firebase.php`
`$config['firebase_app_key'] = __DIR__ . '/../config/firebase_credentials.json';`

Where  does it seek for the credentials?

**Comment by jeromegamez** [comment]
I have no idea what the autoload.php or config array refer to. The error you mentioned only appears when the factory has been initialized without a service account and the autodiscovery kicks in

**Comment by sujeet-agrahari** [comment]
@jeromegamez  It's okay. I will figure it out. Thanks.
